### PR TITLE
Various corrections + Dockerfile

### PR DESCRIPTION
**Pull request by dimon222**
1. Fix npm registry references to point to public NPM
2. Add Dockerfile for easier containerization/local testing for both front-end and back-end
3. Minor restructuring of requirements.txt and conda-forge (it might be wrong, but up for discussion)

How to "docker build" for front-end:
```sh
docker build webapp/frontend -t frontend
```
then run with ` docker run -it -p 127.0.0.1:1234:80 frontend` - accessible as http://localhost:1234 in browser

How to "docker build" for back-end:
```sh
docker build . -f ./webapp/backend/Dockerfile -t backend
```
then run with `docker run -it -p 127.0.0.1:1235:5000 backend`  - accessible as http://localhost:1235 in browser

### CODE REVIEW

File: webapp/backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+ENV FLASK_RUN_HOST=0.0.0.0 \
+	FLASK_RUN_PORT=5000
+WORKDIR /app
+
+COPY ./webapp/backend/requirements.txt requirements.txt
+RUN pip install -r requirements.txt
+COPY mediamanager /app/mediamanager
+COPY ./webapp/backend .
+
+CMD [ "flask", "run" ]
```
Comment (eIGato) [inline webapp/backend/Dockerfile:12]: Please add a line feed.

File: webapp/frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+RUN npm ci
+COPY . /app
+
+RUN node_modules/.bin/ng build
+
+
+FROM docker.io/nginx:1.21.6
+
+COPY --from=build /app/dist /usr/share/nginx/html
+
```
Comment (eIGato) [inline webapp/frontend/Dockerfile:16]: Please remove a line feed.

File: webapp/backend/app.py [hunk 3]
```diff
@@ -16,8 +16,8 @@ def create_app():
     db.init_app(app)
 
     # blueprint for auth routes in our app
-    from webapp.backend.main import main as main_blueprint
-    from webapp.backend.hooks import hooks as hooks_blueprint
+    from main import main as main_blueprint
```
Comment (eIGato) [inline webapp/backend/app.py:19]: `from .main`

Comment (dimon222) [reply webapp/backend/app.py:19]:
```
  File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 77, in find_best_app
    app = app_factory()
  File "/app/app.py", line 19, in create_app
    from .main import main as main_blueprint
ImportError: attempted relative import with no known parent package
```

**Review by andreinechaev — CHANGES_REQUESTED** [review]
`pytest` should work for future CI. 
Current behavior ```E   ModuleNotFoundError: No module named 'webapp'```

**Comment by dimon222** [comment]
> `pytest` should work for future CI. Current behavior `E ModuleNotFoundError: No module named 'webapp'`

How to replicate? As far as I'm aware nothing is present in repo currently for pytest to work.

**Comment by dimon222** [comment]
Do we plan to run pytest globally or is it per each app in repo? Currently its a bit chaotic with parts split on folders, but due to our shared `mediamanager` library it makes things more complicated.

**Comment by dimon222** [comment]
Above comment can be taken as separate conversation for future. Fixed tests for this PR particularly.

**Comment by dimon222** [comment]
PR is obsolete (necessary features are already merged). UI Dockerfile will be contributed separately.
Closed.

**Pull request closed without merging**